### PR TITLE
[LIMINAL-105] add k8s secret cfg

### PR DESCRIPTION
**Pull request by naturalett**
## Changes Title (replace this with a logical title for your changes)

*Issue #, if available:*

### *Description of changes:*

Replace this with the PR description (mention the changes you have made, why you have made them, provide some background and any references to the provider documentation if needed, etc.).

For more information on contributing, please see [Contributing](../CONTRIBUTING.md)
section of our documentation.

### Status

Replace this: describe the PR status. Examples:

- WIP - work in progress
- DONE - ready for review

### Checklist (tick everything that applies)

- [ ] [PreCommitChecks - Code linting](../CONTRIBUTING.md#InstallRunPreCommit) (required)
- [ ] [Tests](../CONTRIBUTING.md#RunningTests)

### CODE REVIEW

File: liminal/kubernetes/secret_util.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+_LOCAL_VOLUMES = set()
+_kubernetes = client.CoreV1Api()
+
+
+def get_secret_configs(liminal_config, base_dir):
+    secrets_config = liminal_config.get('secrets', [])
+
+    for volume_config in secrets_config:
+        if 'secret' in volume_config and 'path' not in volume_config:
+            secret_path = f"{os.getcwd()}/credentials.txt"
```
Comment (assapin) [inline liminal/kubernetes/secret_util.py:46]: what is the purpose of this?
what happens when there are two secretes without a `path` configuration?

File: liminal/kubernetes/secret_util.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    name = conf['secret']
+
+    _LOG.info(f'Requested secret {name}')
+
+    if name not in _LOCAL_VOLUMES:
+        matching_secrets = _kubernetes.list_namespaced_secret(
+            namespace, field_selector=f'metadata.name={name}'
+        ).to_dict()['items']
+
+        while len(matching_secrets) == 0:
```
Comment (assapin) [inline liminal/kubernetes/secret_util.py:70]: why do we need a loop?

File: docs/liminal/kubernetes/secret_util.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+pipelines:
+  - pipeline: k8s_secret_example
+    owner: Bosco Albert Baracus
+    start_date: 1970-01-01
+    timeout_minutes: 10
+    schedule: 0 * 1 * *
+    tasks:
+      - task: my_python_task
+        type: python
+        cmd: aws s3 ls
```
Comment (assapin) [inline docs/liminal/kubernetes/secret_util.md:56]: how does the aws cli know that it needs to do base64 decoding of the 'credentials' entry in the dict?

Comment (naturalett) [reply docs/liminal/kubernetes/secret_util.md:56]: I already decoded it in the secrets

File: liminal/kubernetes/secret_util.py (hunk 1 above)
Comment (naturalett) [reply liminal/kubernetes/secret_util.py]: I fixed the issue with multiple secrets

**Review by assapin — APPROVED** [review]

**Pull request merged**